From 2bebd9d14cbc7f966e4d26ae1e4f04dc3100bd29 Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Thu, 29 Sep 2016 01:02:28 -0400
Subject: [PATCH] always use speed mode ahead-of-time compilation

The interpreter is too slow without the JIT to help it.

Signed-off-by: anupritaisno1 <www.anuprita804@gmail.com>
---
 target/product/runtime_libart.mk | 38 ++++++++++++--------------------
 1 file changed, 14 insertions(+), 24 deletions(-)

diff --git a/target/product/runtime_libart.mk b/target/product/runtime_libart.mk
index ec0f89b81..2916e4b2e 100644
--- a/target/product/runtime_libart.mk
+++ b/target/product/runtime_libart.mk
@@ -95,34 +95,24 @@ PRODUCT_SYSTEM_PROPERTIES += \
 PRODUCT_SYSTEM_PROPERTIES += \
     ro.dalvik.vm.native.bridge=0
 
-# Different dexopt types for different package update/install times.
-# On eng builds, make "boot" reasons only extract for faster turnaround.
-ifeq (eng,$(TARGET_BUILD_VARIANT))
-    PRODUCT_SYSTEM_PROPERTIES += \
-        pm.dexopt.first-boot?=extract \
-        pm.dexopt.boot-after-ota?=extract
-else
-    PRODUCT_SYSTEM_PROPERTIES += \
-        pm.dexopt.first-boot?=verify \
-        pm.dexopt.boot-after-ota?=verify
-endif
-
 # Note that `cmdline` is not strictly needed but it simplifies the management
 # of compilation reason in the platform (as we have a unified, single path,
 # without exceptions).
 PRODUCT_SYSTEM_PROPERTIES += \
-    pm.dexopt.post-boot?=extract \
-    pm.dexopt.install?=speed \
-    pm.dexopt.install-fast?=skip \
-    pm.dexopt.install-bulk?=speed \
-    pm.dexopt.install-bulk-secondary?=verify \
-    pm.dexopt.install-bulk-downgraded?=verify \
-    pm.dexopt.install-bulk-secondary-downgraded?=extract \
-    pm.dexopt.bg-dexopt?=speed \
-    pm.dexopt.ab-ota?=speed \
-    pm.dexopt.inactive?=verify \
-    pm.dexopt.cmdline?=verify \
-    pm.dexopt.shared?=speed
+    pm.dexopt.first-boot=speed \
+    pm.dexopt.boot-after-ota=speed \
+    pm.dexopt.post-boot=speed \
+    pm.dexopt.install=speed \
+    pm.dexopt.install-fast=speed \
+    pm.dexopt.install-bulk=speed \
+    pm.dexopt.install-bulk-secondary=speed \
+    pm.dexopt.install-bulk-downgraded=speed \
+    pm.dexopt.install-bulk-secondary-downgraded=speed \
+    pm.dexopt.bg-dexopt=speed \
+    pm.dexopt.ab-ota=speed \
+    pm.dexopt.inactive=speed \
+    pm.dexopt.cmdline=speed \
+    pm.dexopt.shared=speed
 
 # Pass file with the list of updatable boot class path packages to dex2oat.
 PRODUCT_SYSTEM_PROPERTIES += \
-- 
2.17.1

